File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] a) {
        HashSet<Integer> s = new HashSet<>();
        for (int num : a) {
            s.add(num);
        }
        int maxi = Integer.MIN_VALUE;
        for(int  i : s){
            int cnt=0;
            if(!s.contains(i-1)){
                cnt++;
                int p=i;
                while(s.contains(p+1)){
                    cnt++;
                    p++;
                }
                maxi = Math.max(cnt,maxi);

            }
        }
        return maxi== Integer.MIN_VALUE?0:maxi;
        
    }
}

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        st = set(nums)
        maxi=0
        for x in st:
            cnt=0;
            if(x-1 not in st):
                cnt+=1
                p=x
                while(p+1 in st):
                    cnt+=1
                    p=p+1
            maxi = max(cnt,maxi)
        return maxi



        